api/src/stampy_chat/chat.py

```diff
@@ -29,117 +29,131 @@
+            prompt.append({"role": "user", "content": "Q: " + message["content"]})
             token_count += len(ENCODER.encode("Q: " + message["content"]))
         else:
-            content = cap(message["content"], int(NUM_TOKENS * HISTORY_FRACTION) - token_count)
-
+            content = message["content"]
             # censor all source letters into [x]
             content = re.sub(r"\[[0-9]+\]", "[x]", content)
+            content = cap(content, max_tokens - token_count)
 
-            history_trnc.append({"role": "assistant", "content": content})
+            prompt.append({"role": "assistant", "content": content})
             token_count += len(ENCODER.encode(content))
 
-        if token_count > int(NUM_TOKENS * HISTORY_FRACTION):
+        if token_count > max_tokens:
             break
+    return prompt[::-1]
 
-    prompt.extend(history_trnc[::-1])
-
-
-    question_prompt = "In your answer, please cite any claims you make back to each source " \
-                      "using the format: [a], [b], etc. If you use multiple sources to make a claim " \
-                      "cite all of them. For example: \"AGI is concerning [c, d, e].\"\n\n"
-
-    if mode == "concise":
-        question_prompt += "Answer very concisely, getting to the crux of the matter in as " \
-                "few words as possible. Limit your answer to 1-2 sentences.\n\n"
-
-    elif mode == "rookie":
-        question_prompt += "This user is new to the field of AI Alignment and Safety - don't " \
-                "assume they know any technical terms or jargon. Still give a complete answer " \
-                "without patronizing the user, but take any extra time needed to " \
-                "explain new concepts or to illustrate your answer with examples. "\
-                "Put extra effort into explaining the intuition behind concepts " \
-                "rather than just giving a formal definition.\n\n"
 
-    elif mode != "default": raise ValueError("Invalid mode: " + mode)
+def construct_prompt(query: str, mode: str, history: Prompt, context: List[Block]) -> Prompt:
+    if mode not in PROMPT_MODES:
+        raise ValueError("Invalid mode: " + mode)
 
+    # History takes the format: history=[
+    #     {"role": "user", "content": "Die monster. You don’t belong in this world!"},
+    #     {"role": "assistant", "content": "It was not by my hand I am once again given flesh. I was called here by humans who wished to pay me tribute."},
+    #     {"role": "user", "content": "Tribute!?! You steal men's souls and make them your slaves!"},
+    #     {"role": "assistant", "content": "Perhaps the same could be said of all religions..."},
+    #     {"role": "user", "content": "Your words are as empty as your soul! Mankind ill needs a savior such as you!"},
+    #     {"role": "assistant", "content": "What is a man? A miserable little pile of secrets. But enough talk... Have at you!"},
```

Comment:
huh, where is this from? isn't it copyright infringement or something?

Reply:
dunno. It was already here. But it looks like it's a (very) small part of a play or something, so should be fine as a case of free use